frontend/coprs_frontend/coprs/views/apiv3_ns/json2form.py

```diff
@@ -2,7 +2,19 @@
 from werkzeug.datastructures import MultiDict
 
 
-def get_form_compatible_data(preserve=None):
+def get_form_compatible_data(preserve=None,
+                             comma_separated=None):
+    """
+    Load flask request data from API/CLI calls (typically Python dict uploaded
+    as JSON data or set of uploaded files) and transform it to a dict in
+    WTForms-compatible format; so we eventually can leverage the WTForms
+    validation logic not only for web-UI but also for API.
+
+    Objects of type list() are joined using a separator into one string.
+    The separator string is by default set to one space character, but fields
+    configured in the comma_separated= array are separated using ', ' string.
+    """
```

Comment:
I was kind of shocked that someone is modifying the uploaded list of strings to space-separated string;  I was blaming WTForms, but then I realized we do -> so at least this comment.  But this is not a method I'd like :-(

Reply:
the docstring is greatly appreciated, I was lost when I looked at code while we discussed this

Reply:
Yeah, I am sorry about this. I know how awfully hacky this is.
You expressed my motivations perfectly in the comment. I just didn't know how to do this correctly. 

Maybe we could move the logic to the WTForms side? Looking at the documentation, they seem to be doing similar things with "custom fields"
https://wtforms.readthedocs.io/en/2.3.x/fields/#custom-fields

```python
class TagListField(Field):
    widget = TextInput()

    def _value(self):
        if self.data:
            return u', '.join(self.data)
        else:
            return u''

    def process_formdata(self, valuelist):
        if valuelist:
            self.data = [x.strip() for x in valuelist[0].split(',')]
        else:
            self.data = []
```

Reply:
I'll try to experiment a bit.